### PR TITLE
qt5: optionally provide webkit module

### PR DESCRIPTION
**Pull request by CamJN**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

When Qt dropped the qt-webkit module, they broke the capybara-webkit gem. This option allows you to use the latest Qt (at least up to 5.6.1) and still use capybara-webkit.

https://github.com/Homebrew/homebrew-core/pull/2087 exists but doesn't cover this change, and seems stalled.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Is this module still actively maintained?



> When Qt dropped the qt-webkit module, they broke the capybara-webkit gem. This option allows you to use the latest Qt (at least up to 5.6.1) and still use capybara-webkit.

It feels like a better solution is to rely on an old version of Qt 5.

**Comment by CamJN** [comment]
I don't want to be too negative but I was doing that but homebrew makes it a real pain.

Specifically I need the latest `qt5` for `wireshark`, but and older one for `capybara-webkit`, and I need to link whichever I'm using at the time (can't use both obviously). Huge headache.

**Comment by MikeMcQuaid** [comment]
> homebrew makes it a real pain.

Or: Qt has made this a real pain 😉.

> Is this module still actively maintained?

This is an important question we'd need answered.

**Comment by CamJN** [comment]
Looks like it is: 
https://lists.webkit.org/pipermail/webkit-qt/2016-August/004068.html
http://qtwebkit.blogspot.ca/2016/08/qtwebkit-im-back.html

**Comment by MikeMcQuaid** [comment]
I'm not sure what I think about this because 5.7.0 has been released and there's no accordingly stable version for that. Does this version work with 5.7.0? @DomT4 @ilovezfs: thoughts?

**Comment by DomT4** [comment]
Yeah, this really depends for me on how quickly it is going to get updated. If it's going to become the only thing blocking Qt5 updates in future I'm 👎 on the idea. If we can build the existing resource against 5.7.0 that's probably fine but does need stating.

It's perhaps worth noting that QtWebKit always was an absolute security mess because of how infrequently Qt (and particularly QtWebKit) updates were pushed to release. Anything Webkit that doesn't have a rapid release schedule quickly runs into these problems. I'm wary that going out of our way to include it again, even optionally, is more or less facilitating terrible security practices.

I guess if people know what they're getting into it's fine, but I tend to worry most people don't really care & just want something that "works".

**Comment by MikeMcQuaid** [comment]
> If we can build the existing resource against 5.7.0 that's probably fine but does need stating.

Yes, agreed this is the big question.

**Comment by tdsmith** [comment]
I poked #qtwebkit on Freenode; the mailing list could also be a good source of people who could help us clarify some of the outstanding questions here.

**Comment by annulen** [comment]
Tarball of legacy version, adjusted to Qt 5.7 (roughly the same as 5.6.1 otherwise, so there was no changelog): http://download.qt.io/community_releases/5.7/5.7.0/qtwebkit-opensource-src-5.7.0.tar.xz

Technology preview 3 can be built against Qt 5.2 - 5.7: https://github.com/annulen/webkit/releases/tag/qtwebkit-tp3, but at the moment it needs to be built separately from other Qt modules. We are working on fix at https://codereview.qt-project.org/169257.

**Comment by annulen** [comment]
Staying with Qt 5.5 is a complete non-sense, it has lots of bugs fixed in 5.6 and later versions, and 5.6 is LTS release while 5.5 is already EOL.

File: Formula/qt5.rb [hunk 1]
```diff
@@ -110,6 +116,11 @@ def install
 
     args << "-skip" << "qtwebengine" if build.without? "webengine"
 
+    if build.with? "qtwebkit"
+      (buildpath/"qtwebkit").install resource("qt-webkit")
+      inreplace ".gitmodules", /.*status = obsolete\n((\s*)project = WebKit\.pro)/, "\\1\n\\2initrepo = true"
```
Comment (MikeMcQuaid) [inline Formula/qt5.rb]: Why is this needed?

**Comment by MikeMcQuaid** [comment]
Would be good to use http://download.qt.io/community_releases/5.7/5.7.0/qtwebkit-opensource-src-5.7.0.tar.xz with Qt 5.7.

File: Formula/qt5.rb (hunk 1 above)
Comment (CamJN) [inline Formula/qt5.rb]: Because nowadays Qt chooses which modules to build based on the `.gitmodules` file.

Comment (MikeMcQuaid) [inline Formula/qt5.rb]: Thanks!

**Comment by MikeMcQuaid** [comment]
I see no harm in this for now but if it ends up blocking Qt updates in future we'll need to drop it.
